src/test/java/org/owasp/esapi/reference/HTTPUtilitiesTest.java

```diff
@@ -246,12 +246,12 @@ public void testGetFileUploads() throws IOException {
 			System.err.println("UPLOAD DIRECTORY: " + ESAPI.securityConfiguration().getUploadDirectory());
 			try {
 				List<?> list = ESAPI.httpUtilities().getFileUploads(request4, home);
+				assertTrue( list.size() > 0 );
 				Iterator<?> i = list.iterator();
 				while ( i.hasNext() ) {
 					File f = (File)i.next();
-					System.out.println( "  " + f.getAbsolutePath() );
+					f.delete();
 				}
-				assertTrue( list.size() > 0 );
 			} catch (ValidationException e) {
 				// TODO: This test cases if failing when we upgrade to commons-fileupload;commons-fileupload:1.4 because of a duplicate file error. Need to figure out why.
 				System.out.println("ERROR in testGetFileUploads() request4: " + e.toString());
```

Comment:
I can probably remove this error at this point.

Please note the found condition in the commit message.  May be worth another bug being opened.

Reply:
Looking closer, I think I want to restructure this to a try/finally block to guarantee those files are gone.  That should eliminate any possibility of the files being left on the system between executions.

`public void testGetFileUploads() throws Exception {`

```
List<File> response2 = new ArrayList<>();
try {
	response2 = ESAPI.httpUtilities().getFileUploads(request2, home);
	assertTrue( response2.size() > 0 );
} finally {
	response2.forEach(file -> file.delete());
}	
```

```
List<File> response4 = new ArrayList<>();
try {
    response4 = ESAPI.httpUtilities().getFileUploads(request4, home);
    assertTrue( response4.size() > 0 );
} finally {
    response4.forEach(file -> file.delete());
}
```

1. Files are guaranteed to be deleted if they are created..
2. Any exceptions thrown will escape and kill the test

Reply:
Updates made in https://github.com/ESAPI/esapi-java-legacy/pull/683/commits/776f4a52faf21c583cdbbe79cff1f72086ed204d